src/cljs/snapshot/lumo/repl.cljs

```diff
@@ -1437,20 +1441,27 @@
            (map str (keys special-doc-map))
            (map str (keys repl-special-doc-map))))])))
 
+(def ^{:private true
+       :doc "The symbol detection regex.
+
+This was taken from the reader specification plus tests at the Clojure REPL."}
+  completion-symbol-regex "[a-zA-Z-.<>_*=&?$%!]{1}[a-zA-Z0-9-.<>_*=&?%$!']*")
+
 (defn ^:export get-completions
   [line cb]
   (if-some [js-matches (re-find #"js/(\S*)$" line)]
     (js/$$LUMO_GLOBALS.getJSCompletions line (second js-matches) cb)
     (let [top-level? (boolean (re-find #"^\s*\(\s*[^()\s]*$" line))
-          ns-alias (second (re-find #"\(*(\b[a-zA-Z-.<>*=&?]+)/[a-zA-Z-]*$" line))
-          line-match-suffix (first (re-find #":?([a-zA-Z-.<>*=&?]*|^\(/)$" line))
+          skip-suffix-check? (or (= "(" line) (string/ends-with? line "/") (empty? line))
+          ns-alias (second (re-find (js/RegExp. (str "\\(*(\\b" completion-symbol-regex ")/[a-zA-Z-]*$")) line))
+          line-match-suffix (first (re-find (js/RegExp. (str "^:$|:?(" completion-symbol-regex "|^\\(/)$")) line))
           line-prefix (subs line 0 (- (count line) (count line-match-suffix)))
           completions (reduce (fn [ret item]
                                 (doto ret
                                   (.push (str line-prefix item))))
-                        #js []
-                        (filter #(is-completion? line-match-suffix %)
-                          (completion-candidates top-level? ns-alias)))]
+                              #js []
+                              (filter #(or skip-suffix-check? (is-completion? line-match-suffix %))
```

Comment:
If I'm reading this correctly, whenever `skip-suffix-check?` is `true`, this `filter` is basically useless since every completion candidate goes through it?

Reply:
Yes you read it right